### ISSUE TITLE
--bids-filter-flag file not found error

### ISSUE THREAD

**Issue by hrswearingen** [nipreps/fmriprep#2669]
### What happened?

Hi , I am trying to use the --bids-filter-flag to only process 1 session of my subjects. But when I use the flag `--bids-filter-flag`, I get the error "`error: Path does not exist`."

The path does exist (it is used previously in my command, so I know that docker is able to reference my directories), and I believe my JSON file is accurate.  The JSON file is below.

I was using version 20.2.6, then upgraded to 21.0.0 and am still having the same problem.

### What command did you use?

```shell
fmriprep-docker /Volumes/Pegasus/nstb/rawdata /Volumes/Pegasus/nstb/derivatives participant --participant-label 7017 --bids-filter-file /Volumes/Pegasus/nstb/derivatives/session01.json --fs-license-file /Applications/freesurfer/7.2.0/license.txt --fs-subjects-dir /Volumes/Pegasus/nstb/derivatives/freesurfer -w /Volumes/Pegasus/nstb/scratch
```

### What version of fMRIPrep are you running?

21.0.0

### How are you running fMRIPrep?

Docker

### Is your data BIDS valid?

Yes

### Are you reusing any previously computed results?

FreeSurfer

### Please copy and paste any relevant log output.

```shell
{
   "t1w":{
      "datatype":"anat",
      "session":"01",
      "suffix":"T1w"
   },
   "bold":{
      "datatype":"func",
      "session":"01",
      "suffix":"bold"
   }
}
```

**Comment by effigies** [comment]
I would add `--shell` to your command. If you look at the `RUNNING: ` line that shows up before your shell, you should see a docker volume mount option that looks like `-v <external>:<internal>` and the translated `--bids-filter-file <internal>` section. This may help diagnose the issue.

**Comment by hrswearingen** [comment]
It doesn't seem to be appearing in my shell:

`fmriprep-docker /Volumes/Pegasus/nstb/rawdata /Volumes/Pegasus/nstb/derivatives participant --participant-label 7017 --bids-filter-file /Volumes/Pegasus/nstb/derivatives/session01.json --fs-license-file /Applications/freesurfer/7.2.0/license.txt --fs-subjects-dir /Volumes/Pegasus/nstb/derivatives/freesurfer -w /Volumes/Pegasus/nstb/scratch --shell`

RUNNING: docker run --rm -e DOCKER_VERSION_8395080871=20.10.11 -it -v /Applications/freesurfer/7.2.0/license.txt:/opt/freesurfer/license.txt:ro -v /Volumes/Pegasus/nstb/rawdata:/data:ro -v /Volumes/Pegasus/nstb/derivatives:/out -v /Volumes/Pegasus/nstb/derivatives/freesurfer:/opt/subjects -v /Volumes/Pegasus/nstb/scratch:/scratch --entrypoint=bash nipreps/fmriprep:21.0.0

**Comment by effigies** [comment]
Ah, the problem is just that we don't actually mount it in the wrapper. We can fix that in the next release, but in the meantime, what you can do is take the `docker run` command (from your normal run, not `--shell`), add a `-v /Volumes/Pegasus/nstb/derivatives/session01.json:/tmp/filter.json` before the image name, and replace the `--bids-filter-file` flag with `--bids-filter-file /tmp/filter.json`.

**Comment by hrswearingen** [comment]
Great, thank you! I was able to get it running using this command (I needed to mount the /Applications/freesurfer folder as well):

`docker run -ti --rm -v /Volumes/Pegasus/nstb/rawdata:/data:ro -v /Volumes/Pegasus/nstb/derivatives:/out -v /Volumes/Pegasus/nstb/derivatives/session01.json:/tmp/filter.json -v /Applications/freesurfer/7.2.0/license.txt:/tmp/license.txt  nipreps/fmriprep:latest /data /out participant --participant-label 7017 --fs-license-file /tmp/license.txt --fs-subjects-dir /Volumes/Pegasus/nstb/derivatives/freesurfer/ --bids-filter-file /tmp/filter.json -w /Volumes/Pegasus/nstb/scratch/`

**Comment by effigies** [comment]
Fixed in #2691, will be included in the 21.0.1 release, expected next week.
